doc/custom_nonconformity_score_proof.ipynb

```diff
@@ -0,0 +1,407 @@
+    "\n",
+    "Here $R_{j}^{LOO}$ is signed (it can be positive or negative) so it is added ($+$) in both cases. Besides, $\\frac{\\alpha}{2}$ is used instead of $\\alpha$ since to get the $(1-\\alpha)$ confidence interval, we look for an upper bound of the confidence interval so that $\\frac{\\alpha}{2}$ of the residuals are above and a lower bound of the confidence interval so that $\\frac{\\alpha}{2}$ of the residuals are below.\n",
+    "\n",
+    "Let's consider the first case $ Y_{n+1} > \\hat{q}_{n,\\frac{\\alpha}{2}}^{+}\\{ g(R_{j}^{LOO}, \\hat{\\mu}_{-j}(X_{n+1})) \\} $. We have (because of the properties of $f$ and $g$):\n",
+    "$$\\begin{align}\n",
+    "(1-\\frac{\\alpha}{2})(n+1) & \\leq \\sum_{j=1}^{n} \\mathbb{1}\\{Y_{n+1} > g(R_{j}^{LOO}, \\hat{\\mu}_{-j}(X_{n+1})) \\} \\\\\n",
+    "&= \\sum_{j=1}^{n} \\mathbb{1}\\{ f(Y_{n+1}, \\hat{\\mu}_{-j}(X_{n+1})) > f(g(R_{j}^{LOO}, \\hat{\\mu}_{-j}(X_{n+1})), \\hat{\\mu}_{-i}(X_{n+1})) \\} \\\\\n",
+    "&= \\sum_{j=1}^{n} \\mathbb{1}\\{ f(g(R_{n+1,j}, \\hat{\\mu}_{-j}(X_{n+1})), \\hat{\\mu}_{-j}(X_{n+1})) > R_{j}^{LOO} \\} \\\\\n",
+    "&= \\sum_{j=1}^{n} \\mathbb{1}\\{ R_{n+1,j} > R_{j,n+1}\\} = \\sum_{j=1}^{n} A^{+}_{n+1,j}\n",
+    "\\end{align}$$\n",
+    "and therefore $n+1 \\in S_{+}$, that is, point $n+1$ is a strange point with a high positive residual.\n",
+    "\n",
+    "Similarly, we can conclude the same for strange point with a high negative residual.\n",
+    "$$\\begin{align}\n",
+    "(1-\\frac{\\alpha}{2})(n+1) & \\leq \\sum_{j=1}^{n} \\mathbb{1}\\{Y_{n+1} < g(R_{j}^{LOO}, \\hat{\\mu}_{-j}(X_{n+1})) \\} \\\\\n",
+    "&= \\sum_{j=1}^{n} \\mathbb{1}\\{ f(Y_{n+1}, \\hat{\\mu}_{-j}(X_{n+1})) < f(g(R_{j}^{LOO}, \\hat{\\mu}_{-j}(X_{n+1})), \\hat{\\mu}_{-j}(X_{n+1})) \\} \\\\\n",
+    "&= \\sum_{j=1}^{n} \\mathbb{1}\\{ f(g(R_{n+1,j}, \\hat{\\mu}_{-j}(X_{n+1})), \\hat{\\mu}_{-j}(X_{n+1})) < R_{j}^{LOO}\\} \\\\\n",
+    "&= \\sum_{j=1}^{n} \\mathbb{1}\\{R_{n+1,j} < R_{j,n+1}\\} = \\sum_{j=1}^{n} A^{-}_{n+1,j} = n - \\sum_{j=1}^{n} A^{+}_{n+1,j}\n",
+    "\\end{align}$$\n",
+    "and therefore $n+1 \\in S_{-}$, that is, point $n+1$ is a strange point with a high negative residual.\n",
+    "\n",
+    "Combining with step 2, we have:\n",
+    "$$\\begin{align}\n",
+    "\\mathbb{P}\\{Y_{n+1}>\\hat{q}_{n,\\frac{\\alpha}{2}}^{+}\\{ g(R_{j}^{LOO}, \\hat{\\mu}_{-j}(X_{n+1})) \\} &\\leq \\alpha \\\\\n",
+    "\\mathbb{P}\\{Y_{n+1}<\\hat{q}_{n,\\frac{\\alpha}{2}}^{-}\\{ g(R_{j}^{LOO}, \\hat{\\mu}_{-j}(X_{n+1})) \\} &\\leq \\alpha\n",
+    "\\end{align}$$\n",
+    "\n",
+    "Since any point cannot be simultaneously in $S_{+}$ and $S_{-}$ otherwise $\\alpha \\geq 1$ (because $A^{+}_{i.}\\geq(1-\\frac{\\alpha}{2})(n+1)$ and $(n-A^{+}_{i.})\\geq(1-\\frac{\\alpha}{2})(n+1)$ => $n\\geq2(1-\\frac{\\alpha}{2})(n+1) \\iff \\alpha \\geq \\frac{n+2}{n+1}$), a point is either in none of the two sets or in only one of them.\n",
```

Comment:
a point is either in none of the two sets  -> a point is at most in one of the two sets

Reply:
Done